src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionBuilderExtensions.cs

```diff
@@ -1,16 +1,21 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using Microsoft.AspNetCore.Protocols;
-using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.AspNetCore.Sockets
 {
     public static class ConnectionBuilderExtensions
     {
         public static IConnectionBuilder UseEndPoint<TEndPoint>(this IConnectionBuilder connectionBuilder) where TEndPoint : EndPoint
         {
-            var endpoint = connectionBuilder.ApplicationServices.GetRequiredService<TEndPoint>();
+            var endpoint = (TEndPoint)connectionBuilder.ApplicationServices.GetService(typeof(TEndPoint));
+
+            if (endpoint == null)
+            {
+                throw new InvalidOperationException($"{nameof(EndPoint)} type {typeof(TEndPoint)} is not registered.");
```

Comment:
Is there a test that throws this error and checks for this error message?

Reply:
No. But now we're debating removing end points altogether. I think I might just do that..

Reply:
This is a bigger change than I'd want to make in this PR.